src/Runner.Worker/CompositeStepsRunner.cs

```diff
@@ -0,0 +1,249 @@
+            step.ExecutionContext.Debug($"Starting: {step.DisplayName}");
+
+            // Set the timeout
+            var timeoutMinutes = 0;
+            var templateEvaluator = step.ExecutionContext.ToPipelineTemplateEvaluator();
+            try
+            {
+                timeoutMinutes = templateEvaluator.EvaluateStepTimeout(step.Timeout, step.ExecutionContext.ExpressionValues, step.ExecutionContext.ExpressionFunctions);
+            }
+            catch (Exception ex)
+            {
+                Trace.Info("An error occurred when attempting to determine the step timeout.");
+                Trace.Error(ex);
+                step.ExecutionContext.Error("An error occurred when attempting to determine the step timeout.");
+                step.ExecutionContext.Error(ex);
+            }
+            if (timeoutMinutes > 0)
+            {
+                var timeout = TimeSpan.FromMinutes(timeoutMinutes);
+                step.ExecutionContext.SetTimeout(timeout);
+            }
+
+#if OS_WINDOWS
+            try
+            {
+                if (Console.InputEncoding.CodePage != 65001)
+                {
+                    using (var p = HostContext.CreateService<IProcessInvoker>())
+                    {
+                        // Use UTF8 code page
+                        int exitCode = await p.ExecuteAsync(workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Work),
+                                                fileName: WhichUtil.Which("chcp", true, Trace),
+                                                arguments: "65001",
+                                                environment: null,
+                                                requireExitCodeZero: false,
+                                                outputEncoding: null,
+                                                killProcessOnCancel: false,
+                                                redirectStandardIn: null,
+                                                inheritConsoleHandler: true,
+                                                cancellationToken: step.ExecutionContext.CancellationToken);
+                        if (exitCode == 0)
+                        {
+                            Trace.Info("Successfully returned to code page 65001 (UTF8)");
+                        }
+                        else
+                        {
+                            Trace.Warning($"'chcp 65001' failed with exit code {exitCode}");
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.Warning($"'chcp 65001' failed with exception {ex.Message}");
+            }
+#endif
+
+            try
+            {
+                await step.RunAsync();
+            }
+            catch (OperationCanceledException ex)
+            {
+                if (step.ExecutionContext.CancellationToken.IsCancellationRequested)
```

Comment:
do we need to distinguish between this step timing-out vs the outer step timing-out?

Reply:
i don't think timeout will work properly right now, since the `step.ExecutionContext.SetTimeout(timeout);` is called on the StepRunner for the whole composite action

Reply:
I'll add a TODO so that I remember to do this for `timeout-minutes`